src/dfx/src/config/cache.rs

```diff
@@ -167,114 +151,64 @@ pub fn install_version(v: &str, force: bool) -> DfxResult<PathBuf> {
             .take(12)
             .map(|byte| byte as char)
             .collect();
-        let temp_p = get_bin_cache(&format!("_{}_{}", v, rand_string))
-            .context("Failed to get temporary bin cache path.")?;
-        std::fs::create_dir(&temp_p).with_context(|| {
-            format!(
-                "Failed to create temporary bin cache dir {}.",
-                temp_p.to_string_lossy()
-            )
-        })?;
-
-        let mut binary_cache_assets =
-            util::assets::binary_cache().context("Failed to get asset binary cache.")?;
+        let temp_p = get_bin_cache(&format!("_{}_{}", v, rand_string))?;
+        dfx_core::fs::create_dir_all(&temp_p)?;
+
+        let mut binary_cache_assets = util::assets::binary_cache().map_err(CacheError::ReadBinaryCacheFailed)?;
         // Write binaries and set them to be executable.
-        for file in binary_cache_assets
-            .entries()
-            .context("Failed to get binary cache archive entires.")?
-        {
-            let mut file = file.context("Failed to get binary cache archive entry.")?;
+        for file in binary_cache_assets.entries().map_err(CacheError::ReadBinaryCacheFailed)? {
```

Comment:
Notice that by reusing the same error type here, we lose information about which one happened
```suggestion
        for file in binary_cache_assets.entries().map_err(CacheError::ReadBinaryCacheEntryFailed)? {
```